examples/gpt-2/main.cpp

```diff
@@ -770,25 +842,27 @@ int main(int argc, char ** argv) {
     }
 
     // keep this buffer alive while evaluating the model
-    std::vector<uint8_t> compute_buffer;
+    ggml_backend_buffer_t buf_compute;
 
     struct ggml_allocr * allocr = NULL;
     // allocate the compute buffer
     {
-        allocr = ggml_allocr_new_measure(GGML_MEM_ALIGN);
+         // alignment required by the backend
+        size_t align = ggml_backend_get_alignment(model.backend);
+        allocr = ggml_allocr_new_measure(align);
 
         // create the worst case graph for memory usage estimation
         int n_tokens = std::min(model.hparams.n_ctx, params.n_batch);
         int n_past = model.hparams.n_ctx - n_tokens;
         struct ggml_cgraph * gf = gpt2_graph(model, allocr, n_past, std::vector<gpt_vocab::id>(n_tokens, 0));
 
         // compute the required memory
-        size_t mem_size = ggml_allocr_alloc_graph(allocr, gf) + GGML_MEM_ALIGN;
```

Comment:
I tested doing this CPU-only for sam.cpp and I had to keep this addition of GGML_MEM_ALIGN (I replaced it with "align" from above, but whatever). Otherwise I got an assert later here:
```
            fprintf(stderr, "%s: not enough space in the buffer (needed %zu, largest block available %zu)\n",
                    __func__, size, max_avail);
            GGML_ASSERT(!"not enough space in the buffer");
```
The difference in sizes was either 64 or 48 bytes and the alignment is 64 bytes, so I decided that I need this addition here in order for sam.cpp to work. Initially I thought you remove it here, because you handle it internally in the ggml now, but I had the described above problem.

Reply:
Here is my experiment: https://github.com/YavorGIvanov/sam.cpp/pull/22

Reply:
Should be fixed now, I forgot to add the alignment to the size. The backends should always return aligned buffers, or with enough space to align them, so this shouldn't be necessary anymore.

Reply:
Great. I will remove and verify it tomorrow, but from your change it seems that indeed it should be resolved :)